src/contracts/interfaces/IPaymentCoordinator.sol

```diff
@@ -39,27 +40,12 @@ interface IPaymentCoordinator {
         IERC20[] tokens;
         // cumulative all-time earnings in each token
         uint256[] amounts;
+        //index in the merkle tree
+        uint256 index;
```

Comment:
seems weird for the 'leaf' struct to have the index in it? In my mind that's part of the proof.

Reply:
This is for the parity of the leaf, which determines the order in which the proof elements are hashed with the leaf to verify it.  So the index is the index of the leaf in the entire tree

Reply:
yeah, I'm saying the index isn't _part of the leaf itself_ per se, it's the leaf's position in the tree. So it doesn't seem right for it to be part of the struct to me.

Reply:
ah ok, I mean fair.  Felt it was cleaner, but we can make it another input.

Reply:
I mean I'm OK with calling it LeafAndIndex or something, it just felt technically incorrect to call it a leaf / describe it as the contents of the leaf.